metacat-main/src/main/java/com/netflix/metacat/main/services/MViewService.java

```diff
@@ -35,61 +36,54 @@
      * @param name name of the origin table
      * @return view
      */
-    TableDto create(
-        @Nonnull
-            QualifiedName name);
+    TableDto create(@Nonnull QualifiedName name);
+
+    /**
+     * Create the view and returns the newly created view.
+     * @param name name of the origin table
+     * @param snapshot To snapshot a list of partitions of the table to this view.
+     * @param filter Filter expression string to use
+     * @return view
+     */
+    TableDto createAndSnapshotPartitions(@Nonnull QualifiedName name, boolean snapshot, String filter);
 
     /**
      * Deletes the view and returns the deleted view.
      * @param name name of the view to be deleted
      * @return deleted view
      */
-    TableDto deleteAndReturn(
-        @Nonnull
-            QualifiedName name);
+    TableDto deleteAndReturn(@Nonnull QualifiedName name);
 
     /**
      * Get the view for the given name.
      * @param name name
      * @return view
      */
-    Optional<TableDto> getOpt(
-        @Nonnull
-            QualifiedName name);
+    Optional<TableDto> getOpt(@Nonnull QualifiedName name);
```

Comment:
why is this `getOpt`? seems like an odd name giving the description says you're getting the view for a name?